src/project/mod.rs

```diff
@@ -0,0 +1,94 @@
+use std::collections::HashSet;
+
+use crate::community::CommunityHandle;
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::serde::{Deserialize, Serialize};
+
+pub type ProjectId = usize;
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct ProjectInputs {
+    pub tag: String,
+    pub name: String,
+    pub description: String,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct ProjectMetadata {
+    pub id: ProjectId,
+    pub tag: String,
+    pub name: String,
+    pub description: String,
+    pub owner_community_handles: HashSet<CommunityHandle>,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct Project {
+    pub metadata: ProjectMetadata,
+    /// Configs for project views indexed by their ids and serialized as JSON string
+    pub view_ids: HashSet<String>,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct ProjectViewInputsMetadata {
+    pub project_id: ProjectId,
+    pub kind: String,
+    pub title: String,
+    pub description: String,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct ProjectViewInputs {
+    pub metadata: ProjectViewInputsMetadata,
+    pub config: ProjectViewConfig,
+}
+
+pub type ProjectViewId = String;
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct ProjectViewMetadata {
+    pub id: ProjectViewId,
+    pub project_id: ProjectId,
+    pub kind: String,
+    pub title: String,
+    pub description: String,
+}
```

Comment:
Why introduce `ProjectView*` structs? I feel they are unnecessarily complex, can all view metadata fit into ProjectViewConfig? And each project directly has ProjectViewConfigs instead of reference it by id?

Reply:
I need to reliably know not only id, but all the views metadata on frontend